pyquil/api/_devices.py

```diff
@@ -39,11 +38,12 @@ def list_devices():
     #   "isa":         an ISA object describing the entire device, serialized as a dictionary,
     #   "noise_model": a NoiseModel object describing the entire device, serialized as a dictionary
     # }
+    if connection is None:
+        connection = ForestConnection()
 
-    session = get_session()
-    config = PyquilConfig()
-
-    return sorted(get_json(session, config.forest_url + "/devices")["devices"].keys())
+    session = connection.session
+    url = connection.forest_cloud_endpoint + '/devices'
```

Comment:
`forest_cloud_endpoint` can be `None`

Reply:
how?

Reply:
I added a default

Reply:
Ah indeed, I think you're right. What if my `.qcs_config` file has `url=None`?

Reply:
I guess it would be converted to string `"None"`, maybe, and then the exception would be caught nicely below.